### ISSUE TITLE
Unable to import enclosure

### ISSUE THREAD

**Issue by DrewK-IVZ** [HewlettPackard/POSH-HPOneView#108]
Please fill in as much information as possible to help resolve your issue.

### Expected Behavior
C7000 enclosure added to OneView

Running the command
`Add-HPOVEnclosure -Hostname $Enclosure -EnclosureGroup $EG -Username $oausernanme -Password $oapassword -ApplianceConnection $oneviewConnection -LicensingIntent $LicInt -confirm:$false -Verbose -debug`

### Actual Behavior
Error at console
`Add-HPOVEnclosure : The supplied enclosure state, null, is not valid. `

This is the verbose / debug version
```
VERBOSE: [SEND-HPOVREQUEST] Request Body: {"hostname":"<DNSName>","username":"<Username>","password":"<password>","licensingIntent":"OneViewStandard","force":fa
lse,"enclosureGroupUri":"/rest/enclosure-groups/e9dde030-3e53-4770-81d3-2fb929e46e27","firmwareBaselineUri":null,"forceInstallFirmware":false,"updateFirmwareOn":"Enc
losureOnly","state":null}
VERBOSE: [SEND-HPOVREQUEST] Request: POST https://<FQDN Server>/rest/enclosures
VERBOSE: [SEND-HPOVREQUEST] Request Header 1: Content-Type = application/json
VERBOSE: [SEND-HPOVREQUEST] Request Header 2: Accept = application/json
VERBOSE: [SEND-HPOVREQUEST] Request Header 3: X-API-Version = 200
VERBOSE: [SEND-HPOVREQUEST] Request Header 4: accept-language = en_US
VERBOSE: [SEND-HPOVREQUEST] Request Header 5: accept-encoding = gzip, deflate
VERBOSE: [SEND-HPOVREQUEST] Request Header 6: auth = [*****REDACTED******]
VERBOSE: [SEND-HPOVREQUEST] Request Header 7: Host = ushouonev01vt.corp.amvescap.net
VERBOSE: [SEND-HPOVREQUEST] Net.WebException Error caught
VERBOSE: [SEND-HPOVREQUEST] The remote server returned an error: (400) Bad Request.
VERBOSE: [SEND-HPOVREQUEST] Exception
VERBOSE: [SEND-HPOVREQUEST] Getting Error Response.
VERBOSE: [SEND-HPOVREQUEST] ERROR RESPONSE: {
    "details":  "",
    "errorSource":  null,
    "nestedErrors":  [

                     ],
    "recommendedActions":  [
                               "Specify a correct enclosure state and try again."
                           ],
    "errorCode":  "INVALID_ENCADD_ENCSTATE",
    "data":  {

             },
    "message":  "The supplied enclosure state, null, is not valid."
}

VERBOSE: [SEND-HPOVREQUEST] Response Status: HTTP 400 [Bad Request]
VERBOSE: [SEND-HPOVREQUEST] Response Header: Connection = 
VERBOSE: [SEND-HPOVREQUEST] Response Header: Transfer-Encoding = 
VERBOSE: [SEND-HPOVREQUEST] Response Header: Cache-Control = 
VERBOSE: [SEND-HPOVREQUEST] Response Header: Content-Type = 
VERBOSE: [SEND-HPOVREQUEST] Response Header: Date = 
VERBOSE: [SEND-HPOVREQUEST] Response Header: Server = 
VERBOSE: [SEND-HPOVREQUEST] Response Header: Via = 
VERBOSE: [SEND-HPOVREQUEST] HTTP 400 error caught.
VERBOSE: [NEW-ERRORRECORD] Building ErrorRecord object
Add-HPOVEnclosure : The supplied enclosure state, null, is not valid. 
```

### Steps to reproduce
Running the `Add-HPOVEnclosure` with the company specific variables always generates the error

### Version Information
HPE OneView PowerShell Library Version (`Get-HPOVVersion`): PowerShell Library Version : 2.0.404.0
HPE OneView Appliance Version:   2.00.02-0219211
Output from `$PSVersionTable` on your Windows Host:
Name  : PSVersion
Value : 4.0

Name  : WSManStackVersion
Value : 3.0

Name  : SerializationVersion
Value : 1.1.0.1

Name  : CLRVersion
Value : 4.0.30319.42000

Name  : BuildVersion
Value : 6.3.9600.16406

Name  : PSCompatibleVersions
Value : {1.0, 2.0, 3.0, 4.0}

Name  : PSRemotingProtocolVersion
Value : 2.2

**Comment by ChrisLynchHPE** [comment]
Please use the `-Monitored` parameter switch.  The `HPOneViewStandard` license option needs to be removed from the `-LicenseIntent` parameter value.

**Comment by DrewK-IVZ** [comment]
I went through a few iterations with the response and still cannot get an enclosure imported

Command `Add-HPOVEnclosure -Hostname $Enclosure -EnclosureGroup $EG -Username $oausernanme -Password $oapassword -Monitored -ApplianceConnection $oneviewConnection -confirm:$false -Verbose -debug`

Error `Add-HPOVEnclosure : Parameter set cannot be resolved using the specified named parameters.`

Realized I don't need the enclosure group for monitored.

Command `Add-HPOVEnclosure -Hostname $Enclosure -Username $oausernanme -Password $oapassword -Monitored -ApplianceConnection $oneviewConnection -confirm:$false -Verbose -debug`

Error `Add-HPOVEnclosure : Updating the enclosure firmware is not supported for a monitored enclosure.`

When I look at the verbose output from the second command I see the request sent 
`VERBOSE: [SEND-HPOVREQUEST] Request Body: {"hostname":"<hostname>","username":"<username>","password":"<password>","licensingIntent":"OneViewStandard","force":fa
lse,"enclosureGroupUri":null,"firmwareBaselineUri":null,"forceInstallFirmware":false,"updateFirmwareOn":"EnclosureOnly","state":"Monitored"}
VERBOSE: [SEND-HPOVREQUEST] Request: POST https://<oneViewServer>/rest/enclosures`

It shows force install firmware as false so I don't understand why it would generate an error.

**Comment by ChrisLynchHPE** [comment]
Hmmm…  I will need to look at that further.

**Comment by ChrisLynchHPE** [comment]
Okay, I have this fixed.  I will be publishing a 2.00 update tomorrow.

**Comment by ChrisLynchHPE** [comment]
This is fixed in [Release 2.0.414.0](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v2.0.414.0).

Closing.
